content/swift/concepts/enums/enums.md

```diff
@@ -0,0 +1,147 @@
+
+## Syntax
+
+```pseudo
+enum MyEnum {
+  Enumeration cases go here
+}
+```
+
+To access specific cases in `MyEnum`, the dot operator `.` is used, following the case name:
+
+```pseudo
+MyEnum.case
+```
+
+## Creating an `enum` and Examples
+
+`enum` cases can be defined on separate lines:
+
+```swift
+enum Level {
+  case beginner
+  case intermediate
+  case advanced
+}
+```
+
+`enum` cases can also be written in a single line for simplicity:
+
+```swift
+enum Level {
+  case beginner, intermediate, advanced
+}
+```
+
+### Raw Value
+
+Raw values can be added for each case if a raw [type](https://www.codecademy.com/resources/docs/swift/data-types) is defined after the `enum` name and trailing colon `:`.
+
+```swift
+enum Bishop: String {
+  case novice = "Beginner"
+  case firstJob = "Magician"
+  case secondJob = "Cleric"
+  case thirdJob = "Priest"
+}
+```
+
+The example above displays an `enum`, named `Bishop`, with a raw type of `String` and cases assigned to their respective raw values.
+
+### Associated Value
+
+Associated values can be added to available cases.
+
+```swift
+enum MageAdvancementTree {
+  case novice
+  case firstJob(String)
+  case secondJob(wizard: String)
+  case thirdJob(mage: String)
+  case fourthJob(arch: String)
+}
+
+var bishop = MageAdvancementTree.thirdJob(mage: "Priest")
+
+print(bishop)
+// Output: thirdJob(mage: "Priest")
+```
+
+An `enum` named `MageAdvancementTree` is created and most of its cases have associated values. Then, a `bishop` variable is declared as a `MageAdvancementTree` and sets the `mage` to a string, `"Priest"`.
+
+## Accessing an `enum` and Examples
+
+An enumeration can be accessed with the dot `.` syntax.
+
+```swift
+enum MageAdvancementTree {
+  case novice, firstJob, secondJob, thirdJob, fourthJob
+}
+
+var bishop = MageAdvancementTree.thirdJob
+bishop = .fourthJob
+
+print(bishop)
+// Output: fourthJob
+```
+
+When a variable is declared and set to an enumeration, it can be re-assigned to a different case value with the shortened dot syntax shown above.
+
+### Switch Case
+
+A `switch` statement is another way to access values within an enumeration. In the example below, the `bishop` variable will be evaluated to match `case .fourthJob` because it was previously set to the `fourthJob` value. Access to the values in `enum MageAdvancementTree` is possible becase `bishop` is already set to it.
+
+```swift
+switch bishop {
+  case .novice:
+    print("Beginner")
+  case .firstJob:
+    print("Magician")
+  case .secondJob:
+    print("Cleric")
+  case .thirdJob:
+    print("Priest")
+  case .fourthJob:
+    print("Bishop")
+}
+// Output: Bishop
+```
+
+### Iterative
+
+Notice that `MageAdvancementTree` adopts the [`CaseIterable`](https://www.codecademy.com/resources/docs/swift/protocols) protocol which gives access to the `allCases` property. Using a `for...in` loop, each value within the enumeration can be accessed.
+
+```swift
+enum MageAdvancementTree: CaseIterable {
```

Comment:
@yangc95 maybe I'm overlooking something but Swift gives me this error: `did you mean to specify a raw type on the enum declaration?` when I run this. So I added `String` to this line, like so:
 `enum MageAdvancementTree: String, CaseIterable {` but then when it prints out I don't get the raw values. I made sure I had the switch cases, but I still keep getting: 
```
novice
firstJob
secondJob
thirdJob
fourthJob
```

Reply:
Ah, thanks for catching this. I see that I didn't add the raw values to the enum. I think I'll rename the enum to Bishop and then I can add in those string values which should print. I'll tag you once I make those changes~